domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl

```diff
@@ -16,39 +33,22 @@ DROP TRIGGER IF EXISTS set_${table.tableName}_md_trigger ON ${myuniversity}_${my
 CREATE TRIGGER set_${table.tableName}_md_trigger BEFORE INSERT ON ${myuniversity}_${mymodule}.${table.tableName}
    FOR EACH ROW EXECUTE PROCEDURE ${myuniversity}_${mymodule}.${table.tableName}_set_md();
 
--- on update populate md fields from the creation date and creator fields
-
+-- Trigger for update:
+-- Overwrite createdDate and createdByUserId by the values stored in creation_date and created_by.
 CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.set_${table.tableName}_md_json()
 RETURNS TRIGGER AS $$
-DECLARE
-  createdDate timestamp WITH TIME ZONE;
-  createdBy text ;
-  updatedDate timestamp WITH TIME ZONE;
-  updatedBy text ;
-  injectedId text;
 BEGIN
-  createdBy = NEW.created_by;
-  createdDate = NEW.creation_date;
-  updatedDate = NEW.jsonb->'metadata'->>'updatedDate';
-  updatedBy = NEW.jsonb->'metadata'->>'updatedByUserId';
-
-  if createdBy ISNULL then
-    createdBy = 'undefined';
-  end if;
-  if updatedBy ISNULL then
-    updatedBy = 'undefined';
+  if NEW.creation_date IS NULL then
+    RETURN NEW;
   end if;
-  if createdDate IS NOT NULL then
--- creation date and update date will always be injected by rmb - if created date is null it means that there is no meta data object
--- associated with this object - so only add the meta data if created date is not null -- created date being null may be a problem
--- and should be handled at the app layer for now -- currently this protects against an exception in the db if no md is present in the json
-    injectedId = '{"createdDate":"'||to_char(createdDate,'YYYY-MM-DD"T"HH24:MI:SS.MS')||'" , "createdByUserId":"'||createdBy||'", "updatedDate":"'||to_char(updatedDate,'YYYY-MM-DD"T"HH24:MI:SS.MSOF')||'" , "updatedByUserId":"'||updatedBy||'"}';
-    NEW.jsonb = jsonb_set(NEW.jsonb, '{metadata}' ,  injectedId::jsonb , false);
+
+  NEW.jsonb = jsonb_set(NEW.jsonb, '{metadata,createdDate}', to_jsonb(NEW.creation_date AT TIME ZONE '+0000'));
```

Comment:
seems like there is no need to specify timezone here as creation_date is already saved at 'UTC'